src/adapters/collateralized_simple_interest_loan_terms.ts

```diff
@@ -0,0 +1,125 @@
+
+const TRANSFER_GAS_MAXIMUM = 200000;
+
+export class CollateralizedLoanTerms {
+    private assert: Assertions;
+
+    constructor(web3: Web3, contractsAPI: ContractsAPI) {
+        this.assert = new Assertions(web3, contractsAPI);
+    }
+
+    public packParameters(params: CollateralizedTermsContractParameters): string {
+        this.assertValidParams(params);
+
+        const { collateralTokenIndex, collateralAmount, gracePeriodInDays } = params;
+
+        const collateralTokenIndexShifted = TermsContractParameters.bitShiftLeft(
+            collateralTokenIndex,
+            100,
+        );
+        const collateralAmountShifted = TermsContractParameters.bitShiftLeft(collateralAmount, 8);
+        const gracePeriodInDaysShifted = TermsContractParameters.bitShiftLeft(gracePeriodInDays, 0);
+
+        const baseTenParameters = collateralTokenIndexShifted
+            .plus(collateralAmountShifted)
+            .plus(gracePeriodInDaysShifted);
+
+        return `0x${baseTenParameters.toString(16).padStart(64, "0")}`;
+    }
+
+    public unpackParameters(packedParams: string): CollateralizedTermsContractParameters {
+        this.assert.schema.bytes32("packedParams", packedParams);
+
+        const collateralTokenIndexHex = `0x${packedParams.substr(39, 2)}`;
+        const collateralAmountHex = `0x${packedParams.substr(41, 23)}`;
+        const gracePeriodInDaysHex = `0x${packedParams.substr(64, 2)}`;
+
+        return {
+            collateralTokenIndex: new BigNumber(collateralTokenIndexHex),
+            collateralAmount: new BigNumber(collateralAmountHex),
+            gracePeriodInDays: new BigNumber(gracePeriodInDaysHex),
+        };
+    }
+
+    public assertValidParams(params: CollateralizedTermsContractParameters) {
+        const { collateralTokenIndex, collateralAmount, gracePeriodInDays } = params;
+
+        this.assertCollateralTokenIndexWithinBounds(collateralTokenIndex);
+        this.assertCollateralAmountWithinBounds(collateralAmount);
+        this.assertGracePeriodInDaysWithinBounds(gracePeriodInDays);
+    }
+
+    private assertCollateralTokenIndexWithinBounds(collateralTokenIndex: BigNumber) {
+        // Collateral token index cannot be a decimal value.
+        if (TermsContractParameters.isDecimalValue(collateralTokenIndex)) {
+            throw new Error(CollateralizerAdapterErrors.INVALID_DECIMAL_VALUE());
+        }
+
+        if (collateralTokenIndex.lt(0) || collateralTokenIndex.gt(MAX_COLLATERAL_TOKEN_INDEX_HEX)) {
+            throw new Error(CollateralizerAdapterErrors.INVALID_TOKEN_INDEX(collateralTokenIndex));
+        }
+    }
+
+    private assertCollateralAmountWithinBounds(collateralAmount: BigNumber) {
+        // Collateral amount cannot be a decimal value.
+        if (TermsContractParameters.isDecimalValue(collateralAmount)) {
+            throw new Error(CollateralizerAdapterErrors.INVALID_DECIMAL_VALUE());
+        }
+
+        if (collateralAmount.lt(0)) {
```

Comment:
For reference, BigNumber has a `isNegative()` function for readability -- though admittedly it's the same code.